### DISCUSSION TITLE
How best to update category and budget of transactions ?

### DISCUSSION THREAD

**Discussion by jjauzion** [firefly-iii/firefly-iii#8169] [Firefly III] [answered]
I'm new to Firefly and something bugs me

I have imported last month of transaction from my banks and now want to start putting them in category or budget.
But to do so, I have to click on each transaction, click edit, click category, enter the category, save the update and same for budget. This will take me ages !

Isn't there any easier way ? Like a dropdown directly on the "all transaction" table so I can change category/budget in just 2 click ?

This is a core feature, changing category and budget, there must something I didn't understand. How do you guys do it ?

**Comment by BackSeat** [comment]
Just define a rule...

**Comment by JC5** [comment]
There is no "why". This is just how Firefly III has happened to have been developed. You think I sit here in my evil lair making it difficult for YOU on purpose?

If you can tone down the passive aggressiveness I'm happy to discuss things but this isn't how it works.

**Comment by jjauzion** [comment]
Sorry if my question was feeling "aggressive", it was not my intention at all. I edit out the "why" into a "how" which indeed make the question more smooth ;)

My question is really to understand if there is something I miss because I feel like editing the category and budget of all transaction one by one is going to take me very long.

> Just define a rule

I can give this a try, I thought rules would only work for 20% or 30% of the transactions which are recurring every month but having a second look at the data I can probably cover about 80% of transactions

But still there is maybe 20 transactions per month that I will have to manually edit each month which can take times if I have to open each transaction individually. Is there a better way of doing it ?

**Reply by wazabees** [reply, marked as answer]
You can edit in bulk or a table view, or am I misunderstanding the question?

https://github.com/firefly-iii/firefly-iii/issues/4238#issuecomment-755980436

**Reply by jjauzion** [reply]
yes ! That's exactly what I was looking for, thanks :)
